k8_cortx_cloud/deploy-cortx-cloud.sh

```diff
@@ -260,6 +260,20 @@ buildValues() {
             | \$to" "${values_file}" "${solution_yaml}"
     done
 
+    # shellcheck disable=SC2016
+    yq -i eval-all '
+        select(fi==0) ref $to | select(fi==1) ref $from
+        | with($to.configmap;
+            .cortxControl.agent.resources           = $from.solution.common.resource_allocation.control.agent.resources
+            | .cortxHa.fault_tolerance.resources    = $from.solution.common.resource_allocation.ha.fault_tolerance.resources
+            | .cortxHa.health_monitor.resources     = $from.solution.common.resource_allocation.ha.health_monitor.resources
+            | .cortxHa.k8s_monitor.resources        = $from.solution.common.resource_allocation.ha.k8s_monitor.resources
+            | .cortxHare.hax.resources              = $from.solution.common.resource_allocation.hare.hax.resources
+            | .cortxMotr.motr.resources             = $from.solution.common.resource_allocation.data.motr.resources
+            | .cortxMotr.confd.resources            = $from.solution.common.resource_allocation.data.confd.resources
+            | .cortxRgw.rgw.resources               = $from.solution.common.resource_allocation.server.rgw.resources)
+        | $to' "${values_file}" "${solution_yaml}"
```

Comment:
Alternatively to updating `solution-check.yaml` to require these values, you could make them optional. I couldn't find a good way in yq to say "if $from.X is defined, then assign $to.X to $from.X", so I came up with an expression that uses the chart's values.yaml file to fill in defaults, and merge them with the solution value. It works, but it's pretty convoluted, so you might find fixing the check file easier.

```suggestion
    yq -i eval-all '
        select(fi==0) ref $to | select(fi==1) ref $from | select(fi==2) ref $base
        | with($to.configmap;
            .cortxControl.agent.resources        = ($base.configmap.cortxControl.agent.resources * $from.solution.common.resource_allocation.control.agent.resources // {})
            | .cortxHa.fault_tolerance.resources = ($base.configmap.cortxHa.fault_tolerance.resources * $from.solution.common.resource_allocation.ha.fault_tolerance.resources // {})
            | .cortxHa.health_monitor.resources  = ($base.configmap.cortxHa.health_monitor.resources * $from.solution.common.resource_allocation.ha.health_monitor.resources // {})
            | .cortxHa.k8s_monitor.resources     = ($base.configmap.cortxHa.k8s_monitor.resources * $from.solution.common.resource_allocation.ha.k8s_monitor.resources // {})
            | .cortxHare.hax.resources           = ($base.configmap.cortxHare.hax.resources * $from.solution.common.resource_allocation.hare.hax.resources // {})
            | .cortxMotr.motr.resources          = ($base.configmap.cortxMotr.motr.resources * $from.solution.common.resource_allocation.data.motr.resources // {})
            | .cortxMotr.confd.resources         = ($base.configmap.cortxMotr.confd.resources * $from.solution.common.resource_allocation.data.confd.resources // {})
            | .cortxRgw.rgw.resources            = ($base.configmap.cortxRgw.rgw.resources * $from.solution.common.resource_allocation.server.rgw.resources // {}))
        | $to' "${values_file}" "${solution_yaml}" ../charts/cortx/values.yaml
```
The `x // {}` defines the default value of `{}` if the solution value is null or unspecified. The map merging doesn't work if one side is `null`.

Reply:
You could also go back to the manual method of extracting the value to a variable and writing it only if not empty.